R/make-array.R

```diff
@@ -80,6 +80,101 @@ makeMR <- function (subvariables, name, selections, ...) {
     return(vardef)
 }
 
+
+#' Create Multiple Response Variable from Delimited
+#'
+#' Surveys often record multiple response questions in delimited lists where
+#' each respondent's selections are separated by a delimiter like `;` or `|`.
+#' This function breaks the delimited responses into subvariables, uploads those
+#' subvariables to Crunch, and finally creates a multiple response variable from
+#' them.
+#'
+#' @param var The variable containing the delimited responses
+#' @param delim The delimiter separating the responses
+#' @param name The name of the resulting MR variable
+#' @param selected A character string used to indicate a selection
+#' @param not_selected Character string identifying non-selection
+#' @param unanswered Character string indicating non-response
+#' @param ... Other arguments to be passed on to [makeMR()]
+#'
+#' @return
+#' @export
+mrFromDelim <- function(var,
+    delim,
+    name,
+    selected = "selected",
+    not_selected = "not_selected",
+    unanswered = NA,
+    ...) {
+    if (missing(name)) {
+        halt("Must supply a name for the new variable")
+    }
+    if (is.Categorical(var) || is.Text(var)) {
+        v <- as.vector(var)
+    } else {
+        halt(dQuote("var"), " must be a Categorical or Text Crunch Variable.")
+    }
+    uniques <- unique(v[!is.na(v)])
+    cats <- unique(unlist(strsplit(uniques, delim)))
+    vardefs <- lapply(cats, function(x) createSubvarDef(v, x, delim,
+        selected, not_selected, unanswered, missing = is.na(v)))
+    ds <- loadDataset(datasetReference(var))
+    addVariables(ds, vardefs)
```

Comment:
How about just changing the subvariable names to `varname_selection`? In other words if the delimited variable was called `trees` then the subvariables would be `trees_oak` `trees_maple` etc.  My preference would be to keep the function consistent so that it always named things in the same way and didn't change behaviour based of the other variables in a dataset.

Reply:
`varname_selection` is nice, but it doesn't ensure uniqueness. A safer way is to create the variable in one request (not upload and then bind, which is not atomic). And in doing it in a single request, you may be able to omit aliases for the subvariables, in which case the backend should generate (guaranteed valid) aliases for them.